src/uwtools/rocoto.py

```diff
@@ -182,13 +182,56 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
-        for key, block in config.items():
-            tag, _ = self._tag_name(key)
-            if tag == STR.taskdep:
-                self._set_attrs(SubElement(e, STR.taskdep), block)
+        for tag, block in config.items():
+            tag, _ = self._tag_name(tag)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
```

Comment:
The same collection exists -- here as a `list`, there as a `tuple` -- in `__add_task_dependency_operator()`. It would be good to DRY that out as either a private class constant or a private `@property` method. Same for the other categories just below.